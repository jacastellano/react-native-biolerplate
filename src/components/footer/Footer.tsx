import * as React from 'react';
import { Footer, FooterTab, Button, Text, Icon } from 'native-base';

interface Props {

}

export const FooterComponent = (props: Props) => {
    return (

        <Footer>
            <FooterTab>
                <Button active={true} onPress={() => { }}>
                    <Icon active={true} name="apps" />
                    <Text>Item1</Text>
                </Button>
                <Button active={false} onPress={() => { }}>
                    <Icon active={true} name="camera" />
                    <Text>Item2</Text>
                </Button>
                <Button active={false} onPress={() => { }}>
                    <Icon active={true} name="compass" />
                    <Text>Item3</Text>
                </Button>
                <Button active={false} onPress={() => { }}>
                    <Icon active={true} name="contact" />
                    <Text>Item4</Text>
                </Button>
            </FooterTab>
        </Footer>
    );
}