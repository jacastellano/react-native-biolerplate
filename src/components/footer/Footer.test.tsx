import * as React from 'react';
import renderer from 'react-test-renderer';

import { FooterComponent } from '../Footer/Footer';

it('renders correctly', () => {
    const footer = renderer
        .create(<FooterComponent />)
        .toJSON();
    expect(footer).toMatchSnapshot();
});