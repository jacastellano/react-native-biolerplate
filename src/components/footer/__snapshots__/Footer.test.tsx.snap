// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F8F8F8",
      "borderColor": "#cbcbcb",
      "borderTopWidth": 0.5,
      "elevation": 3,
      "flexDirection": "row",
      "height": 55,
      "justifyContent": "center",
      "left": 0,
      "paddingBottom": 0,
      "right": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": undefined,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#cde1f9",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flex": 1,
          "flexDirection": null,
          "height": 55,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        active={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#007aff",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#007aff",
            "fontFamily": "System",
            "fontSize": 14,
            "lineHeight": 16,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        Item1
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flex": 1,
          "flexDirection": null,
          "height": 55,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        active={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#6b6b6b",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#6b6b6b",
            "fontFamily": "System",
            "fontSize": 14,
            "lineHeight": 16,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        Item2
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flex": 1,
          "flexDirection": null,
          "height": 55,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        active={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#6b6b6b",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#6b6b6b",
            "fontFamily": "System",
            "fontSize": 14,
            "lineHeight": 16,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        Item3
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": null,
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flex": 1,
          "flexDirection": null,
          "height": 55,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
        }
      }
    >
      <Text
        active={true}
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#000",
                "fontSize": 30,
              },
              Object {
                "color": "#6b6b6b",
                "fontSize": 24,
                "marginLeft": 16,
                "marginRight": 16,
                "paddingTop": 2,
              },
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#6b6b6b",
            "fontFamily": "System",
            "fontSize": 14,
            "lineHeight": 16,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        Item4
      </Text>
    </View>
  </View>
</View>
`;
